modal/cli/dict.py

```diff
@@ -0,0 +1,123 @@
+from modal_proto import api_pb2
+
+dict_cli = Typer(
+    name="dict",
+    no_args_is_help=True,
+    help="Manage `modal.Dict` objects and inspect their contents.",
+)
+
+
+@dict_cli.command(name="create")
+@synchronizer.create_blocking
+async def create(name: str, *, env: Optional[str] = ENV_OPTION):
+    """Create a named Dict object."""
+    d = _Dict.from_name(name, environment_name=env, create_if_missing=True)
+    client = await _Client.from_env()
+    resolver = Resolver(client=client)
+    await resolver.load(d)
+
+
+@dict_cli.command(name="list")
+@synchronizer.create_blocking
+async def list(*, json: bool = False, env: Optional[str] = ENV_OPTION):
+    """List all named Dict objects."""
+    env = ensure_env(env)
+    client = await _Client.from_env()
+    request = api_pb2.DictListRequest(environment_name=env)
+    response = await retry_transient_errors(client.stub.DictList, request)
+
+    def format_timestamp(t: float) -> str:
+        return datetime.strftime(datetime.fromtimestamp(t), "%Y-%m-%d %H:%M") + " UTC"
+
+    rows = [(d.name, format_timestamp(d.created_at)) for d in response.dicts]
+    display_table(["Name", "Created at"], rows, json)
+
+
+@dict_cli.command("clear")
+@synchronizer.create_blocking
+async def clear(name: str, *, env: Optional[str] = ENV_OPTION):
+    """Clear the contents Dict by deleting all of its data."""
+    d = await _Dict.lookup(name, environment_name=env)
+    await d.clear()
+
+
+@dict_cli.command(name="delete")
+@synchronizer.create_blocking
+async def delete(name: str, *, env: Optional[str] = ENV_OPTION):
+    """Delete a named Dict object and all of its data."""
+    client = await _Client.from_env()
+    lookup_request = api_pb2.AppGetByDeploymentNameRequest(
+        name=name,
+        environment_name=ensure_env(env),
+        namespace=api_pb2.DEPLOYMENT_NAMESPACE_WORKSPACE,
+    )
+    resp = await client.stub.AppGetByDeploymentName(lookup_request)
+    stop_req = api_pb2.AppStopRequest(app_id=resp.app_id, source=api_pb2.APP_STOP_SOURCE_CLI)
+    await client.stub.AppStop(stop_req)
+
+
+@dict_cli.command(name="get")
+@synchronizer.create_blocking
+async def get(name: str, key: str, *, env: Optional[str] = ENV_OPTION):
+    """Print the value for a specific key.
+
+    Note: When using the CLI, keys are always interpreted as having a string type.
+    """
+    d = await _Dict.lookup(name, environment_name=env)
+    console = Console()
+    val = await d.get(key)
+    console.print(val)
+
+
+@dict_cli.command(name="show")
+@synchronizer.create_blocking
+async def show(
+    name: str,
+    n: int = Argument(default=None, help="Retrieve and show no more than this many entries"),
+    *,
+    json: bool = False,
+    env: Optional[str] = ENV_OPTION,
+):
+    """Print the contents of a Dict.
+
+    Note: By default, this command will retrieve the complete Dict contents.
+    Using the N argument is recommended for quickly inspecting a small number of items.
```

Comment:
I think it is not great that the default behavior could potentially grab the contents of a huge Dict. But also having a default non-null value for `N` feels like it could be confusing too. I have considered also adding a `modal dict peek` command that has such a default, which makes sense to me. Then people's muscle memory could point them towards the cheap operation. Open to adding if it sounds like a good idea to others.

Reply:
seems fine to have a default number for n if we print something like `... 584 more values not shown`

Reply:
We can print something that indicates truncation although currently our estimate of dict size is very wrong so I don't think we can say exactly how many :)

Reply:
Also as written there is no way to invoke `modal dict show` in a way that *doesn't* truncate ... how would you spell that, `modal dict show -1`? That seems kind of weird.

Reply:
`modal dict show --unlimited` or something? idk

Reply:
Went with `modal dict show --all` and a default size of `20`